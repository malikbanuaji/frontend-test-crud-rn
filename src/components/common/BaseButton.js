import React, {useCallback} from 'react';
import {StyleSheet, Text, TouchableNativeFeedback, View} from 'react-native';
import {Spacing, Typography, Mixins} from '../../styles';

export default function BaseButton({
  text,
  onPress,
  style,
  textStyle,
  disabled,
}) {
  const handlePress = useCallback(() => {
    if (disabled !== true) {
      onPress();
    }
  }, [disabled, onPress]);

  return (
    <View style={[style, disabled && styles.disabled]}>
      <TouchableNativeFeedback disabled={disabled} onPress={handlePress}>
        <View style={styles.button}>
          <Text style={[styles.buttonText, textStyle]}>{text}</Text>
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonText: {
    ...Typography.buttonText,
  },
  button: {
    padding: Spacing.smaller,
  },
  disabled: {
    ...Mixins.disabled,
  },
});
