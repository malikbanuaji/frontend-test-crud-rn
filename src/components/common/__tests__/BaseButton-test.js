import React from 'react';

import {fireEvent, render} from '@testing-library/react-native';
import BaseButton from '../BaseButton';

test('renders correctly', () => {
  const tree = render(<BaseButton text={'MockedText'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when disabled', () => {
  const tree = render(
    <BaseButton disabled={true} text={'MockedText'} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('handle press correctly', () => {
  const mockOnPress = jest.fn((x) => 42 + x);
  const {getByText} = render(
    <BaseButton onPress={mockOnPress} text={'MockedText'} disabled={false} />,
  );
  fireEvent.press(getByText('MockedText'));
  expect(mockOnPress).toBeCalled();
});

test('handle press correctly when disabled', () => {
  const mockOnPress = jest.fn();
  const {getByText} = render(
    <BaseButton onPress={mockOnPress} text={'MockedText'} disabled={true} />,
  );
  fireEvent.press(getByText('MockedText').parent.parent);
  expect(mockOnPress).not.toBeCalled();
});
