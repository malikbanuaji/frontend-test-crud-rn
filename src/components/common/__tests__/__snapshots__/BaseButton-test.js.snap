// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      MockedText
    </Text>
  </View>
</View>
`;

exports[`renders correctly when disabled 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "opacity": 0.5,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      MockedText
    </Text>
  </View>
</View>
`;
